### ISSUE TITLE
Seriously Slow LFO bug

### ISSUE THREAD

**Issue by martin-lueders** [almostEric/FrozenWasteland#15]
Currently, the seriously slow LFO get's stuck, e.g. at 44100kHz sample rate and times larger 12 Minutes.
This is due to the fact that the internal phase is just a float and underflows in this cases.

This can be fixed by changing the phase to a double.

However, there is another little problem which might cause grieve when trying to fix it.
The two versions of the seriously slow LFO both declare their own LFO structure, which is global. If you change phase to double in only one of them, things start going crazy. You should either make these structs private, or use the same struct (class) in both versions of the seriously slow LFO.

I fixed the problem for me by deleting the extremly slow version as I have not found any use for that so far.

**Comment by almostEric** [comment]
Hi Martin,

 

I’ve never had an issue with the seriously slow LFO getting stuck (I’ve used it at time bases of several hours), but it certainly doesn’t hurt to make it more accurate, so I have changed the phase and delta phase variables to doubles. I also made the LFO structs private.

 

The CDC signature version of the seriously slow LFO was pretty much a joke, and I haven’t ported it to 0.6, and have no plans to unless someone asks.

 

Thanks for your input

 

Cheers.

Eric
